File: prob1.py
# Time Complexity : O(n)
# Space Complexity :O(n)
# Passed on Leetcode: yes

class Solution:
    def numberOfArithmeticSlices(self, nums: List[int]) -> int:
        n = len(nums)

        if n < 3:
            return 0
        dp = [0 for _ in range(n)]
        res = 0
        for i in range(n-3, -1, -1):
            if nums[i] - nums[i+1] == nums[i+1] - nums[i+2]:
                dp[i] = 1 + dp[i+1]
            else:
                dp[i] = 0
            
            res += dp[i]

        return res

File: prob1.java
// Time Complexity : O(n^2)  
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes

class Solution {
    public int numberOfArithmeticSlices(int[] nums) {
	
        if(nums.length<3) return 0;
        int count=0;
        for(int i=0 ; i<nums.length-2 ; i++)
        {
            if(i+2<nums.length && nums[i+1]-nums[i] == nums[i+2]-nums[i+1])
            {
                int d=nums[i+1] - nums[i];
                count++;
				
                for(int j=i+2;j<nums.length-1 && nums[j+1]-nums[j]==d;j++) count++;
            }
        }
        return count;
    }
}